pkg/kustomize/kustomize.go

```diff
@@ -492,16 +492,24 @@ func Inflate(ctx *quaycontext.QuayRegistryContext, quay *v1.QuayRegistry, baseCo
 		return nil, err
 	}
 
+	created := v1.GetCondition(quay.Status.Conditions, v1.ConditionComponentsCreated)
+	migrationInProgress := created != nil && created.Reason == v1.ConditionReasonMigrationsInProgress
+	log.Info("----------------------------------------------------------------------------------")
+	log.Info(fmt.Sprintf("migrationInProgress is: '%t'", migrationInProgress))
+	log.Info(fmt.Sprintf("currentVersion is: '%s'", quay.Status.CurrentVersion))
+
 	var overlay string
 	if rolloutBlocked(quay) {
 		overlay = configEditorOnlyOverlay()
-	} else if quay.Status.CurrentVersion != v1.QuayVersionCurrent {
+	} else if quay.Status.CurrentVersion != v1.QuayVersionCurrent && !migrationInProgress {
```

Comment:
I guess it is safe to assume that this part of the code won't be executed if the migrations are running (ConditionReasonMigrationsInProgress). No need to check for `!migrationInProgress`.

Reply:
Hmm yes, true!
Additionally, `quay.Status.CurrentVersion != v1.QuayVersionCurrent && !migrationInProgress` will evaluate to true if the migration job has failed, since we change the condition reason to migration failed.
Given that we set `spec.restartPolicy: OnFailure` in the upgrade job, I don't think we ever want to recreate the job except when there are new migrations to run.
We should only recreate the upgrade job (to force the job to run again) when the following is true:
1. `QUAY_VERSION` differs from `status.currentVersion`
2. the current job has run successfuly, so `status.succeeded == 1`
What you think?

Reply:
Looks like a made a wrong assumption here. The upgrade job overlay will not recreate the job, it'll just set quay-app replicas to 0.
I think this code can stay as is.